src/main/java/build/buildfarm/instance/shard/ShardInstance.java

```diff
@@ -432,6 +452,35 @@ public void run() {
     } else {
       operationQueuer = null;
     }
+
+    prometheusMetricsThread =
+        new Thread(
+            () -> {
+              while (true) {
+                try {
+                  TimeUnit.SECONDS.sleep(30);
+                  OperationsStatus operationsStatus = operationsStatus();
+                  workerPoolSize.set(operationsStatus.getActiveWorkersCount());
+                  dispatchedOperations.set(operationsStatus.getDispatchedSize());
+                  preQueueSize.set(operationsStatus.getPrequeue().getSize());
+                  updateQueueSizes(operationsStatus.getOperationQueue().getProvisionsList());
+                } catch (InterruptedException e) {
```

Comment:
If `Thread.stop()` is used, there won't be any `InterruptedException`, but the thread get killed directly. 

Considering `stop()` is deprecated (see my comment below), we might want to use `Thread.interrupt()`. If that's the case, the behavior of the thread might be different from our expectation: If the current thread is interrupted when it's sleeping (i.e. ` TimeUnit.SECONDS.sleep(30)`, it will throw InterruptedException, which is fine. However, if it's not sleeping, there wont be any execution, we have to do check the status of `Thread.currentThread.isInterrupted()` to know if the thread should be stopped. If ignore it, the thread wont exit.

Reply:
Good catch, I see we are using stop() in other places as well - this may be a good story to replace all of them in the same PR. Not really sure if it matters in this use case (same with DispatchedMonitor and other threads) as in this case the main process is already stopping so we just want to shut everything down.

Reply:
Yeah, since we are shutting everything done, the problems of `Thread.stop()` are not really concerns. We don't have to deal with it right now.